CUJExecutor/dataAggregation/AggregateExecuteCuj.java

```diff
@@ -0,0 +1,173 @@
+		}
+		formattedTable.add(medianRow);
+
+		return formattedTable;
+	}
+
+
+
+	/**
+	 * parses the passed inputfile and prints a summary of the data, which includes the total number of iterations executed,
+	 * and a table with the durations of each action (and the total) for each iteration, as well as averages and medians for
+	 * each action (and the total). Each column is also labeled by the token used in the file passed to ./executeCUJ and
+	 * a note is printed detailing how many iterations of data are missing (if any are).
+	 *
+	 *
+	 * @param inputFile - location of a file that contains the data from each iteration through the CUJ executed by ./executeCuj
+	 * 		      file must be in the following form:
+	 * 		     
+	 * 		      d1
+	 *	              t1
+	 *	              d2
+	 *	              t2
+	 *
+	 *	              .
+	 *	              .
+	 *	              .
+	 *
+	 *	              dn
+	 *	              tn
+	 *
+	 *	              Where the executeCuj call successfully iterated through the CUJ n times. Each di must be a comma-seperated
+	 *	              list of integers, where each integer corresponds to the duration of an action in the CUJ (passed sequentially).
+	 *	              Each ti must consist of two comma-seperated times in the form HH:MM:SS.sss, where the first corresponds to the time 
+	 *	              when this iteration started and the second time corresponds to the time when this iteration ended (relative to the 
+	 *	              beginning of the video containing every iteration). An example file is provided at 
+	 *	              dataAggregation/mockFiles/mockExecuteCujOutput.txt.
+	 *	             
+	 * @param outputStream - the location of the file where we should dump the summary table produced by parsing the given data
+	 * @param postCUJ - a comma-seperated list of actions of the form specified by the ./executeCuj script
+	 * @param totalIters - the total number of iterations expected to have been executed
+	 * @param clipDestinationFolder - the folder where we should store the output clip if one is produced
+	 * @param trim - "true" or "false", telling us whether we should produce a trimmed clip
+	 */
+	public static void main (String[] args) throws Exception {
+
+		//Prepare inputs/outputs
+		File inputFile = new File(args[0]);
+		Scanner myReader = new Scanner(inputFile);
+		OutputStream outputStream = new FileOutputStream(args[1]);
+		PrintStream ps = new PrintStream(outputStream);
+		String headerFooter = "\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n";
+		for (String s : args) {
+			System.out.println(s);
+		}
+
+		String[] postCUJ = args[2].split("\\s*,\\s*");
+		int totalIters = Integer.parseInt(args[3]);
+		String clipDestinationFolder = args[4];
+		boolean trim = Boolean.parseBoolean(args[5]); //tell us whether we want to trim the video
+
+		List<List<Integer>> actionDurations = new ArrayList<>(totalIters);//2D list : j = test index, k = action index
```

Comment:
I'm using a 2D list to hold measurements (and later a 3D list in megaData.java) and a map from total durations to clipInfo objects to later recover the clipInfo object corresponding to the median totalDuration. In retrospect, it would be cleaner to have an object representing an iteration (holding a clipInfo and a list of action durations). This would lower the dimensionality of the actionDurations lists, eliminate the need for a map, and yield operations that make more sense than things like "calculationg the average/median of columns". Do you think it would be worthwhile to make this change?

Reply:
Yes, almost always better to use POJOs (plain-old-java-objects to present complex data).

Reply:
Great! (already started implementing this)